csharp_dotnetcore/3.Welcome-Users/WelcomeUserBot.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Newtonsoft.Json;
+using WelcomeUser.State;
+
+namespace WelcomeUser
+{
+    /// <summary>
+    /// Represents a bot that can process incoming activities.
+    /// For each interaction from the user, an instance of this class is called.
+    /// For each Activity received, a new instance of this class is created.
+    /// Objects that are expensive to construct, or have a lifetime.
+    /// beyond the single Turn, should be carefully managed and chached - see Startup.cs.
+    /// </summary>
+    public class WelcomeUserBot : IBot
+    {
+        // generic message sent to user
+        private const string _genericMessage = "This is a simple Welcome Bot sample.You can say \'intro\' to see the introduction card. If you are running this bot in the Bot Framework emulator, press the \'Start Over\' button to simulate user joining a bot or a channel";
+
+        // the bot state accessor object. Use this to access specific state properties
+        private readonly WelcomeUserStateAccessors _welcomeUserStateAccessors;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WelcomeUserBot"/> class.
+        /// Initalize WelcomeUserBot.
+        /// </summary>
+        /// <param name="statePropertyAccessor"> Bot state accessor object.</param>
+        public WelcomeUserBot(WelcomeUserStateAccessors statePropertyAccessor)
+        {
+            _welcomeUserStateAccessors = statePropertyAccessor ?? throw new System.ArgumentNullException("state accessor can't be null");
+        }
+
+        /// <summary>
+        /// Every Conversation turn for our WelcomeUser Bot will call this method, including
+        /// any type of activiaties such as ConversationUpdate or ContactRelationUpdate which
+        /// are sent when a user joins a conversation.
+        /// This bot doesn't use any dialogs; it's "single turn" processing, meaning a single
+        /// request and response, with no stateful conversation between user input.
```

Comment:
Incorrect comment. This bot *DOES* store state to track if a welcome message has been sent.